### ISSUE TITLE
learning rate for train

### ISSUE THREAD

**Issue by Sirius-Li** [KaiChen1998/GeoDiffusion#17]
I find the learning rate in the train script is 1.5e-4, different from "4e−5 for U-Net and 3e−5 for the text encoder" in paper. I also try some different learning rates, finding that lr between 5e-5 and 1.5e-4 is ok, the loss plots are similar, what should the learning rate set?

**Comment by KaiChen1998** [comment]
- For `512x512` and `800x456` experiments, the default LR for UNet and text encoder is 4e-5 and 3e-5 respectively with a batch size of 64.
- For `256x256` experiments, due to the decreased GPU memory requirement, we can use a batch size of 32x8=256 with 8 V100 GPUs, and the linearly-scaled LR is `(4e-5)x(32x8/64)=1.6e-4` for UNet.
- However, due to the large batch size optimization, we find `LR=1.5e-4` achieves a little bit better empirical results with 8 GPUs.
- For training GeoDiffusion, the `--learning_rate` argument controls the LR for UNet, while the `--lr_text_ratio` argument determines the ratio between LRs of UNet and text encoder.
- For example, `--learning_rate=4e-5` and `--lr_text_ratio=0.75` suggest the LR for UNet is 4e-5, while the LR for text encoder is 4e-5x0.75=3e-5.

**Comment by Sirius-Li** [comment]
Alright, thanks a lot for your help!!!

**Comment by alion-code** [comment]
> * 对于 和 experiments，UNet 和文本编码器的默认 LR 分别为 4e-5 和 3e-5，批处理大小为 64。`512x512``800x456`
> * 对于实验，由于 GPU 内存需求降低，我们可以使用 32x8=256 的批量大小和 8 个 V100 GPU，线性缩放的 LR 适用于 UNet。`256x256``(4e-5)x(32x8/64)=1.6e-4`
> * 然而，由于大批量优化，我们发现使用 8 个 GPU 取得了更好的实证结果。`LR=1.5e-4`
> * 对于训练 GeoDiffusion，参数控制 UNet 的 LR，而参数确定 UNet 的 LR 与文本编码器之间的比率。`--learning_rate``--lr_text_ratio`
> * 例如，并建议 UNet 的 LR 为 4e-5，而文本编码器的 LR 为 4e-5x0.75=3e-5。`--learning_rate=4e-5``--lr_text_ratio=0.75`

Hello KaiChen1998, why is the maximum batch size I can set only 2 when training with 256*256 resolution images on a V100 using this code? @KaiChen1998

**Comment by KaiChen1998** [comment]
@alion-code Hi! Thank you for your interest in GeoDiffusion!

I'm not quite sure what the problem is. If it's right with you, you can also show us your code for further checking.

There are several potential problems you might double-check:

1. Follow the hyperparameters in our provided [training script](https://github.com/KaiChen1998/GeoDiffusion/blob/main/tools/dist_train.sh). For example, remember to use FP16 or BF16 for mixed-precision training and SDv1.5 as the base model.

2. Make sure you are really training with 256x256 resolution. The default batch size for 512x512 is indeed 2 for V100.

3. Make sure you are using the same dependencies with [ours](https://github.com/KaiChen1998/GeoDiffusion/tree/main?tab=readme-ov-file#installation). Newer versions of dependencies might affect the GPU memory requirement spike.

**Comment by alion-code** [comment]
> [@alion-code](https://github.com/alion-code) Hi! Thank you for your interest in GeoDiffusion!
> 
> I'm not quite sure what the problem is. If it's right with you, you can also show us your code for further checking.
> 
> There are several potential problems you might double-check:
> 
> 1. Follow the hyperparameters in our provided [training script](https://github.com/KaiChen1998/GeoDiffusion/blob/main/tools/dist_train.sh). For example, remember to use FP16 or BF16 for mixed-precision training and SDv1.5 as the base model.
> 2. Make sure you are really training with 256x256 resolution. The default batch size for 512x512 is indeed 2 for V100.
> 3. Make sure you are using the same dependencies with [ours](https://github.com/KaiChen1998/GeoDiffusion/tree/main?tab=readme-ov-file#installation). Newer versions of dependencies might affect the GPU memory requirement spike.

Thank you for your reply. I found that I was using the wrong version of accelerate. After reinstalling accelerate==0.17.1, I can now train on a V100 with a batch size of 4, which occupies about 28GB of GPU memory. Thank you very much!

**Comment by KaiChen1998** [comment]
@alion-code 

Hi! I'm glad to know that, and if possible, you might also want to align the versions of the other dependencies with ours (e.g., the PyTorch version may also play a significant role), which could further enhance training efficiency.

Feel free to re-open this issue or open new issues if you have any other questions!
